packages/aws-cdk/lib/import.ts

```diff
@@ -0,0 +1,362 @@
+      const props = fmtdict(fixedIdInput);
+
+      if (!await promptly.confirm(
+        `${chalk.blue(resourceName)} (${resourceType}): import with ${chalk.yellow(props)} (yes/no) [default: yes]? `,
+        { default: 'yes' },
+      )) {
+        print(chalk.grey(`Skipping import of ${resourceName}`));
+        return undefined;
+      }
+    }
+
+    // Ask the user to provide missing props
+    const userInput: ResourceIdentifierProperties = {};
+    for (const missingIdProp of missingIdProps) {
+      const response = (await promptly.prompt(
+        `${chalk.blue(resourceName)} (${resourceType}): enter ${chalk.blue(missingIdProp)} to import (empty to skip):`,
+        { default: '', trim: true },
+      ));
+      if (!response) {
+        print(chalk.grey(`Skipping import of ${resourceName}`));
+        return undefined;
+      }
+      userInput[missingIdProp] = response;
+    }
+
+    return {
+      ...fixedIdInput,
+      ...userInput,
+    };
+  }
+
+  /**
+   * Convert the internal "resource mapping" structure to CloudFormation accepted "ResourcesToImport" structure
+   */
+  private async makeResourcesToImport(resourceMap: ImportMap): Promise<ResourcesToImport> {
+    return resourceMap.importResources.map(res => ({
+      LogicalResourceId: res.logicalId,
+      ResourceType: res.resourceDiff.newResourceType!,
+      ResourceIdentifier: resourceMap.resourceMap[res.logicalId],
+    }));
+  }
+
+  /**
+   * Convert CloudFormation logical resource ID to CDK construct tree path
+   *
+   * @param logicalId CloudFormation logical ID of the resource (the key in the template's Resources section)
+   * @returns Forward-slash separated path of the resource in CDK construct tree, e.g. MyStack/MyBucket/Resource
+   */
+  private describeResource(logicalId: string): string {
+    return this.stack.template?.Resources?.[logicalId]?.Metadata?.['aws:cdk:path'] ?? logicalId;
+  }
+}
+
+/**
+ * Information about a resource in the template that is importable
+ */
+export interface ImportableResource {
+  /**
+   * The logical ID of the resource
+   */
+  readonly logicalId: string;
+
+  /**
+   * The resource definition in the new template
+   */
+  readonly resourceDefinition: any;
+
+  /**
+   * The diff as reported by `cloudformation-diff`.
+   */
+  readonly resourceDiff: ResourceDifference;
+}
+
+/**
+ * The information necessary to execute an import operation
+ */
+export interface ImportMap {
+  /**
+   * Mapping logical IDs to physical names
+   */
+  readonly resourceMap: ResourceMap;
+
+  /**
+   * The selection of resources we are actually importing
+   *
+   * For each of the resources in this list, there is a corresponding entry in
+   * the `resourceMap` map.
+   */
+  readonly importResources: ImportableResource[];
+}
+
+export function mkdict<A>(xs: Array<readonly [string, A]>): Record<string, A> {
+  const ret: Record<string, A> = {};
+  for (const [k, v] of xs) {
+    ret[k] = v;
+  }
+  return ret;
+}
```

Comment:
(Optional/Note) This looks awkwardly like `Object.fromEntries` (https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries)